android/src/main/java/com/marigold/rnsdk/RNMarigoldModule.kt

```diff
@@ -0,0 +1,685 @@
+
+import android.app.Activity
+import android.content.Intent
+import android.location.Location
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.WritableNativeArray
+import com.marigold.sdk.model.AttributeMap
+import com.marigold.sdk.Marigold
+import com.marigold.sdk.MessageStream
+import com.marigold.sdk.enums.ImpressionType
+//import com.marigold.sdk.model.ContentItem
+import com.marigold.sdk.model.Message
+import com.marigold.sdk.MessageActivity
+import com.marigold.sdk.model.Purchase
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.marigold.sdk.EngageBySailthru
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.reflect.InvocationTargetException
+import java.net.URI
+import java.net.URISyntaxException
+import java.util.ArrayList
+import java.util.Date
+
+/**
+ * React native module for the Marigold SDK.
+ */
+class RNMarigoldModule(reactContext: ReactApplicationContext, private val displayInAppNotifications: Boolean) : ReactContextBaseJavaModule(reactContext), MessageStream.OnInAppNotificationDisplayListener {
+
+    companion object {
+        const val ERROR_CODE_DEVICE = "marigold.device"
+        const val ERROR_CODE_MESSAGES = "marigold.messages"
+        const val ERROR_CODE_RECOMMENDATIONS = "marigold.recommendations"
+        const val ERROR_CODE_TRACKING = "marigold.tracking"
+        const val ERROR_CODE_VARS = "marigold.vars"
+        const val ERROR_CODE_PURCHASE = "marigold.purchase"
+        const val MESSAGE_ID = "id"
+        fun setWrapperInfo() {
+            try {
+                //val setWrapperMethod = Marigold::class.java.getDeclaredMethod("setWrapper", String::class.java, String::class.java)
+                val cArg = arrayOf(String::class.java, String::class.java)
+//                val setWrapperMethod = Marigold::class.java.getDeclaredMethod("setWrapper", *cArg)
+
+                val companionClass = Marigold.Companion::class.java
+                val setWrapperMethod = companionClass.getDeclaredMethod("setWrapper", *cArg)
+
+                setWrapperMethod.isAccessible = true
+                setWrapperMethod.invoke(Marigold.Companion, "React Native", "10.0.0")
+            } catch (e: NoSuchMethodException) {
+                e.printStackTrace()
+            } catch (e: IllegalAccessException) {
+                e.printStackTrace()
+            } catch (e: InvocationTargetException) {
+                e.printStackTrace()
+            }
+        }
+    }
+
+    @VisibleForTesting
+    var marigold = Marigold()
+
+    @VisibleForTesting
+    var engage = EngageBySailthru()
```

Comment:
This will fail and throw errors for apps not using the Sailthru product. I think we should create a separate `RNEngageBySailthruModule.kt` module for the ST specific functionality. In the new module, rather than keeping an instance we should add a helper method to create the instance each time it's needed, and surface any returned errors as appropriate.